web/packages/shared/hooks/useAsync.ts

```diff
@@ -93,8 +93,9 @@ export function useAsync<Args extends unknown[], AttemptData>(
   const run = useCallback(
     (...args: Args) => {
       setState(prevState => ({
-        ...prevState,
         status: 'processing',
+        data: prevState['data'],
+        statusText: prevState['statusText'],
```

Comment:
Here I wanted to remove spread in favor of explicitly setting the fields. This shouldn't change the behavior as `Attempt` used to have those three fields only.